cirq-google/cirq_google/calibration/workflow_test.py

```diff
@@ -1367,6 +1375,66 @@ def test_run_zeta_chi_gamma_calibration_for_moments() -> None:
     assert calibrated_circuit.moment_to_calibration == [None, None, 0, None, None, 1, None]
 
 
+def test_make_zeta_chi_gamma_compensation_for_moments() -> None:
+    parameters_ab = cirq_google.PhasedFSimCharacterization(zeta=0.5, chi=0.4, gamma=0.3)
+    parameters_bc = cirq_google.PhasedFSimCharacterization(zeta=-0.5, chi=-0.4, gamma=-0.3)
+    parameters_cd = cirq_google.PhasedFSimCharacterization(zeta=0.2, chi=0.3, gamma=0.4)
+
+    a, b, c, d = cirq.LineQubit.range(4)
+
+    circuit = cirq.Circuit(
+        [
+            [cirq.X(a), cirq.Y(c)],
+            [cirq_google.SycamoreGate().on(a, b), cirq.CZ.on(c, d)],
+            [SQRT_ISWAP_INV_GATE.on(b, c)],
+        ]
+    )
+
+    options = cirq_google.FloquetPhasedFSimCalibrationOptions(
+        characterize_theta=False,
+        characterize_zeta=True,
+        characterize_chi=True,
+        characterize_gamma=True,
+        characterize_phi=False,
+    )
+
+    circuit_with_calibration = workflow.CircuitWithCalibration(circuit, [None, 0, 1])
```

Comment:
Added this check.
It forced me to rewrite `test_zeta_chi_gamma_calibration_for_moments`, because now each gate has its own engine gate, and only that engine gate can be passed in `characterizations`.